### DISCUSSION TITLE
Middle click on Notification applet to turn off notification [Do not distrub mode]

### DISCUSSION THREAD

**Discussion by b4kspace** [linuxmint/discussions#510] [Ideas]
This will reduce the unnecessary four clicks to just one middle click and fulfill the request to have a 'do not disturb' mode feature

**Comment by rcalixte** [comment]
There are already options for this:

* There is a default applet you can add called `inhibit@cinnamon.org` that supports key bindings and a reduces the required number of clicks.
* There is also a third-party applet you can install with the Applets module called [Notifications Enhanced](https://cinnamon-spices.linuxmint.com/applets/view/393). This is a clone of the default notifications applet with the additional toggle for enabling/disabling notifications. Middle-clicking sounds like it would be a great feature request for this applet.

**Reply by b4kspace** [reply]
Thank you for the reply, @rcalixte. I knew about the inhibit applet but not the keybindings. Wouldn't it make more sense to have the keybindings in the default Notification and Power applets? I still believe middle-click muting would enhance the usability of default applet and eliminate the need for additional applets. but ultimately, it's up to the developers to decide if they want to keep the current functionality or implement this improvement.

**Comment by igorsantos07** [comment]
Middle click in an applet is not a standard interaction and would be odd, to say the least. The user would need to discover this interaction exist and remember about it to start using. The option should definitely be included in the usual right-click menu as well.
>Edit: by reading the 21.3 release notes I discovered the sound applet already have a middle click action to mute or play/plause, which is configurable on its settings. So I guess I am wrong on the "non-standard" factor, albeit the discovery factor is still there - but I understand it could be useful for power users.

Besides that, IMHO I found the idea of "basic applets offering very basic features and then having clones of them offering more options" a very bad experience. You end up with duplicated work and the third-party developer having to sync with the original add-on, if they want to keep up with the original style.
These developers should be invited to implement those features in the main applet, instead of people being suggested to install third-party stuff when they request a feature (with the potential of being abandoned in the future).
>I'm not criticising rcalixte's response, but the overall behavior which seems to be growing of having third-party, enhanced copies of extensions. So far, his suggestion is _indeed_ the best one, but doesn't seem like a healthy idea for the Cinnamon community in the long run.

Bottom line: given a standard and existing applet already have middle-click capabilities, it would make a lot of sense if the standard notification applet offered a similar feature as well - even more if you consider both middle-clicks would be used for muting.

**Reply by Hilyxx** [reply]
@igorsantos07 I suggested to implement those features in the main applet but my request was rejected. Consequently I've made "Notification-Enhanced" applet.
